### ISSUE TITLE
[Rendering] .NET Templating 1.3.1 creates invalid card if string value contains json

### ISSUE THREAD

**Issue by justin-mellor** [microsoft/AdaptiveCards#7570]
### Target Platforms

Other

### SDK Version

2.7.3 and 1.3.1

### Application Name

Bot Framework

### Problem Description

AdaptiveCards.Templating 1.3.1 now creates invalid json for the card if the object being passed contains a string representation of json. This worked in 1.3.0

WIth 1.3.0 the output was
`{"type":"AdaptiveCard","$schema":"http://adaptivecards.io/schemas/adaptive-card.json","version":"1.3","body":[{"id":"stringWithJson","type":"TextBlock","text":"String With JSON - {\"number1\":23}","wrap":true}]}
`
With 1.3.1 the output is
`{"type":"AdaptiveCard","$schema":"http://adaptivecards.io/schemas/adaptive-card.json","version":"1.3","body":[{"id":"stringWithJson","type":"TextBlock","text":"String With JSON - {"number1":23}","wrap":true}]}`

Basically the string has not been encoded

### Card JSON

```JSON
{
    "type": "AdaptiveCard",
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.3",
    "body": [
        {
            "id": "stringWithJson",
            "type": "TextBlock",
            "text": "String With JSON - ${$root.stringWithJson}",
            "wrap": true
        }
    ]
}
```

### Sample Code Language

C#

### Sample Code

```
var template = new AdaptiveCards.Templating.AdaptiveCardTemplate(cardJson);

var card = template.Expand(new { stringWithJson = @"{""number1"":23}" });
```

**Comment by jwoo-msft** [comment]
> @justin-mellor - looks like you can get the behavior you want by using a template like this (note `jsonStringify` below):
> 
> ```json
> {
>     "type": "AdaptiveCard",
>     "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
>     "version": "1.3",
>     "body": [
>         {
>             "id": "stringWithJson",
>             "type": "TextBlock",
>             "text": "String With JSON - ${jsonStringify($root.stringWithJson)}",
>             "wrap": true
>         }
>     ]
> }
> ```
> 
> @jwoo-msft can give more context here :)




> This seems to me a breaking change in the templating functionality. Also if the data comes from user entered data then you have no way to know if the user happens to enter json. I don't see how this is not a bug. @jwoo-msft - Can you provide an explanation




> @justin-mellor - looks like you can get the behavior you want by using a template like this (note `jsonStringify` below):
> 
> ```json
> {
>     "type": "AdaptiveCard",
>     "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
>     "version": "1.3",
>     "body": [
>         {
>             "id": "stringWithJson",
>             "type": "TextBlock",
>             "text": "String With JSON - ${jsonStringify($root.stringWithJson)}",
>             "wrap": true
>         }
>     ]
> }
> ```
> 
> @jwoo-msft can give more context here :)




> This seems to me a breaking change in the templating functionality. Also if the data comes from user entered data then you have no way to know if the user happens to enter json. I don't see how this is not a bug. @jwoo-msft - Can you provide an explanation
@justin-mellor,
 
if the data is serialized json, then you're correct. if the data is in deserialized form of json, then you have to use jsonStringify. 
Either way, if we are talking about serialized json, the functionality was broken. It is fixed in #7578.

**Comment by justin-mellor** [comment]
Great Joseph. It was about serialized json. Thanks
